pilot/pkg/model/push_context.go

```diff
@@ -1637,6 +1637,10 @@ func (ps *PushContext) SetDestinationRules(configs []config.Config) {
 
 	for i := range configs {
 		rule := configs[i].Spec.(*networking.DestinationRule)
+		// filter the destination rule which belong to root namespace and with short name host, such as 'review'
+		if configs[i].Namespace == ps.Mesh.RootNamespace && rule.Host != "" && host.IsShortNameForHost(rule.Host) {
```

Comment:
What is the impact of this change on users? This feels like a breaking API change?

Reply:
Hi @howardjohn, this code change is located in `SetDestinationRules` for initiation of DestinationRules, and this change means that it will not handle the destination rule with namespace is root namespace and short name host as we discussed above comments.
Moreover, I did not find any break except the test case, that's why I fixed the test case to add ps mesh root namespace.

Reply:
> What is the impact of this change on users? This feels like a breaking API change?

Hi @howardjohn, the impact of this change on users is that the DR will not be handled if short name as host when DR is deployed to **istio-system** according to the comment here: https://github.com/istio/istio/pull/36402#discussion_r764037691. However, user still can create such DR but it will be ignored. 
I can remove this process in **SetDestinationRules** and just keep the analyzer parts if you don't want to break API change. Does it make sense?

Reply:
What was the old behavior? Was it adding FQDN with the root namespace suffix?

Reply:
yes, @nrjpoddar, based on the issue 36237, I have debug that DestinationRule is in `istio-system` and VirtualService is in `default`, moreover,  DR and VS has short name host in specified subsets. Then, this host in DR has been interpreted as  `reviews.istio-system.xxx.yyy` and  in VS has been interpreted as `reviews.default.xxx.yyy` .  So this is the root cause why there is an error for not found: "hostname+version-v3" like this.

Reply:
> A: The old behavior is that DestinationRule is in istio-system and VirtualService is in default, moreover, DR and VS has short name host in specified subsets. Then, this host in DR has been interpreted as reviews.istio-system.xxx.yyy and in VS has been interpreted as reviews.default.xxx.yyy. So this is the root cause why there is an error for not found: 'hostname+version-v3' like this.


That sounds like what I would expect. isn't this user error?

Reply:
> > A: The old behavior is that DestinationRule is in istio-system and VirtualService is in default, moreover, DR and VS has short name host in specified subsets. Then, this host in DR has been interpreted as reviews.istio-system.xxx.yyy and in VS has been interpreted as reviews.default.xxx.yyy. So this is the root cause why there is an error for not found: 'hostname+version-v3' like this.
> 
> That sounds like what I would expect. isn't this user error?

Hi @howardjohn, yes, it is, so the error message says that `reviews.default.xxx.yyy` is not defined in DR is correct. But according to the comment [here](https://github.com/istio/istio/pull/36402#discussion_r764037691): short name as host when DR is deployed to istio-system should not be allowed,  I submitted the latest code change.

Reply:
I don't agree with that comment. We *cannot* make backwards incompatible changes to our API.

If its a "bad idea", we can add a warning. But not an error.

Reply:
> I don't agree with that comment. We _cannot_ make backwards incompatible changes to our API.
> 
> If its a "bad idea", we can add a warning. But not an error or change behavior.

okay, if so I suggest that let's come back to my previous commit and keep the original logic and add a warning message for analysis and warning message for API parts. I will post a new submit for it.

Reply:
Hi @howardjohn ,  done!
1. There is a warning message for user to recommend them to provide the whole host for destination rule which is belonged to root namespace
2. There is no error message when doing analysis, but a warning message instead